src/tpm2/crypto/openssl/CryptEccSignature.c

```diff
@@ -616,6 +618,63 @@ BnValidateSignatureEcdsa(
  Exit:
     return retVal;
 }
+#else // USE_OPENSSL_FUNCTIONS_ECDSA     libtpms added begin
+TPM_RC
+BnValidateSignatureEcdsa(
+			 bigNum                   bnR,           // IN: r component of the signature
+			 bigNum                   bnS,           // IN: s component of the signature
+			 bigCurve                 E,             // IN: the curve used in the signature
+			 //     process
+			 bn_point_t              *ecQ,           // IN: the public point of the key
+			 const TPM2B_DIGEST      *digest         // IN: the digest that was signed
+			 )
+{
+    ECDSA_SIG        *sig = NULL;
+    EC_KEY           *eckey = NULL;
+    int               retVal;
+    int               rc;
+    BIG_INITIALIZED(r, bnR);
```

Comment:
The use of these macros is a bit concerning, as they completely mask any errors that may happen.
If any of the numbers fail to initialize we may give an avenue for an attacker to probe the signature scheme. I am not sure there is an attack, but I would rather fail early than risk getting through private key operations with potentially known (0 point) or even attacker controlled (stack values) input.

Reply:
Using `BIGNUM *r = BigInitialized(bnR);` now.